### PR TITLE
Fix Android tests and add them to CI

### PR DESCRIPTION
**Pull request by shridharkalagi**
## Change list

1. Fixing the android tests
2. Add Android CI integration

## Types of changes

What types of changes are you proposing/introducing to Java client?
_Put an `x` in the boxes that apply_

- [x] No changes in production code.
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Details
Have Included few tests in CI which always pass on the latest version on Android Emulator available on travis (android-25)

### CODE REVIEW

File: src/test/java/io/appium/java_client/android/UIAutomator2Test.java [hunk 1]
```diff
@@ -5,106 +5,84 @@
...
 import org.openqa.selenium.support.ui.ExpectedConditions;
 import org.openqa.selenium.support.ui.WebDriverWait;
 
-import java.io.File;
-import java.util.concurrent.TimeUnit;
 
-public class UIAutomator2Test {
-    private static AppiumDriverLocalService service;
-    protected static AndroidDriver<AndroidElement> driver;
+public class UIAutomator2Test extends BaseAndroidTest {
```
Comment (SrinivasanTarget) [inline src/test/java/io/appium/java_client/android/UIAutomator2Test.java]: this class can be deleted and all the tests can be moved to `AndroidDriverTest`.

File: build.gradle [hunk 2]
```diff
@@ -228,5 +228,9 @@ task uiAutomationTest( type: Test ) {
     useJUnit()
     testLogging.showStandardStreams = true
     testLogging.exceptionFormat = 'full'
-    include '**/UIAutomationTest.class'
+    filter {
+        includeTestsMatching '*.SettingTest'
+        includeTestsMatching 'io.appium.java_client.android.ClipboardTest'
+        includeTestsMatching '*.AndroidAppStringsTest'
```
[the remark this answers is not in the document]
Comment (shridharkalagi) [reply build.gradle]: Few tests run only on real devices, few needs adb to run as root and few are flaky (depends on the android emulator). Hence, included only those which are consistently running. Other tests can be picked once travis supports latest Emulator.

File: .travis.yml [hunk 3]
```diff
@@ -14,33 +13,42 @@ matrix:
...
+      android:
+        components:
+          - extra-android-support
+          - sys-img-armeabi-v7a-android-24
+      before_install:
+        - curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.33.1/install.sh | bash
+        - nvm install 8
+        - node --version
+        - npm --version
+        - npm install -g appium
```
[the remark this answers is not in the document]
Comment (shridharkalagi) [reply .travis.yml]: Ok Will update this. Does it need the below PR to be merged?

https://github.com/appium/java-client/pull/1001

[the remark this answers is not in the document]
Comment (SrinivasanTarget) [reply .travis.yml]: Nope. You can use right away.

File: src/test/java/io/appium/java_client/android/UIAutomator2Test.java (hunk 1 above)
Comment (shridharkalagi) [reply src/test/java/io/appium/java_client/android/UIAutomator2Test.java]: `AndroidDriverTest` is not running in CI currently. Can this be done later?

Comment (shridharkalagi) [reply src/test/java/io/appium/java_client/android/UIAutomator2Test.java]: @SrinivasanTarget ping

Comment (SrinivasanTarget) [reply src/test/java/io/appium/java_client/android/UIAutomator2Test.java]: Ok make travis happy

**Comment by shridharkalagi** [comment]
@SrinivasanTarget CI is failing as iOS tests not running due to missing env variables. 
 https://travis-ci.org/appium/java-client/builds/421407707

File: .travis.yml [hunk 4]
```diff
@@ -1,3 +1,4 @@
+language: java
```
Comment (SrinivasanTarget) [inline .travis.yml]: remove this its not required.

**Comment by SrinivasanTarget** [comment]
@shridharkalagi It's not failing due to env variables. I will have a look.

**Comment by shridharkalagi** [comment]
@SrinivasanTarget can you check this pls?

**Comment by saikrishna321** [comment]
@SrinivasanTarget fix conflicts pls

**Comment by shridharkalagi** [comment]
Fixed the conflicts. @SrinivasanTarget iOS CI is failing. Can you have a look?

**Comment by SrinivasanTarget** [comment]
@shridharkalagi Travis is fine now. Make Codacy happy.

**Pull request merged**